service/iothub-service-sdk/src/main/java/com/microsoft/azure/iot/service/transport/amqps/AmqpSend.java

```diff
@@ -111,9 +111,14 @@ public void send(String deviceId, Message message) throws IOException
             // Codes_SRS_SERVICE_SDK_JAVA_AMQPSEND_12_007: [The function shall initialize the Proton reactor object]
             this.reactor = Proton.reactor(this);
             // Codes_SRS_SERVICE_SDK_JAVA_AMQPSEND_12_008: [The function shall start the Proton reactor object]
-            this.reactor.run();
+            try
+            {
+              this.reactor.run();
+            }
+            catch (Exception e) {
```

Comment:
I think it would be nice to throw exception to higher layers and notify then to catch it here.